core/dnsserver/config_test.go

```diff
@@ -0,0 +1,122 @@
+package dnsserver
+
+import (
+	"github.com/coredns/caddy"
+	"testing"
+)
+
+func Test_keyForConfig(t *testing.T) {
+	type args struct {
+		blocIndex    int
+		blocKeyIndex int
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "Test #1",
+			args: args{
+				blocIndex:    0,
+				blocKeyIndex: 0,
+			},
+			want: "0:0",
+		},
+		{
+			name: "Test #2",
+			args: args{
+				blocIndex:    0,
+				blocKeyIndex: 1,
+			},
+			want: "0:1",
+		},
+		{
+			name: "Test #3",
+			args: args{
+				blocIndex:    1,
+				blocKeyIndex: 0,
+			},
+			want: "1:0",
+		},
+		{
+			name: "Test #4",
+			args: args{
+				blocIndex:    1,
+				blocKeyIndex: 1,
+			},
+			want: "1:1",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := keyForConfig(tt.args.blocIndex, tt.args.blocKeyIndex); got != tt.want {
+				t.Errorf("keyForConfig() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetConfig(t *testing.T) {
+	type args struct {
+		keyIndex   int
+		BlockIndex int
+	}
+	tests := []struct {
+		name string
+		args args
+		want *Config
+	}{
+		{
+			name: "Test #1",
```

Comment:
These test cases don't seem to be testing useful corner cases and instead test boolean permutation of non-boolean variables (`int`).

pro tip: use description of a test as a test name (e.g `"return nil on invalid key"`). If you can't think of a good description probably not a useful test then.

Reply:
The idea was to set the 
 ```
 c.ServerBlockKeyIndex = tt.args.keyIndex
c.ServerBlockIndex = tt.args.BlockIndex
```
So that `c` can be passed to `GetConfig`. If there are other corner cases please let me know and I can fix them.

Thanks